### PR TITLE
Model usage validation hook added

### PR DESCRIPTION
**Pull request by dhrn**
The model can be validated and checked it is usage by passing `validateUsage=true` query param as part the request

### CODE REVIEW

File: activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/main/java/org/activiti/cloud/services/modeling/service/ModelServiceImpl.java [hunk 1]
```diff
@@ -450,17 +451,24 @@ public void validateModelContent(Model model,
...
+        ValidationContext validationContext = getValidationContext(model, fileContent, null);
         validateModelContent(model.getType(),
                              fileContent.getFileContent(),
                              validationContext);
     }
 
 
 
+    private ValidationContext getValidationContext(Model model, FileContent fileContent, @Nullable Project project) {
+        if (!modelTypeService.isJson(findModelType(model)) && fileContent.getContentType().equals(CONTENT_TYPE_JSON))
```
Comment (erdemedeiros) [inline activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/main/java/org/activiti/cloud/services/modeling/service/ModelServiceImpl.java]: Please, always use brackets for the body of an `if` statement, even if it contains only a single line.

File: activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/main/java/org/activiti/cloud/services/modeling/service/ModelServiceImpl.java [hunk 2]
```diff
@@ -450,17 +451,24 @@ public void validateModelContent(Model model,
...
+            return EMPTY_CONTEXT;
+
+        if(project != null)
```
Comment (erdemedeiros) [inline activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/main/java/org/activiti/cloud/services/modeling/service/ModelServiceImpl.java]: Same here: brackets missing.

**Review by erdemedeiros — CHANGES_REQUESTED** [review]
Generally speaking, I'm not a big fun of boolean flags as they tend to increase the complexity of the code. However, I'm ok to keep them here to avoid a bigger refactoring for now. There are only two small fixes and it's ok to merge.

**Review by erdemedeiros — APPROVED** [review]

**Comment by dhrn** [comment]
@Mergifyio refresh

**Pull request merged**